ajax/libs/color-hash.js/1.0.3/color-hash.min.js

```diff
@@ -0,0 +1 @@
+(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.ColorHash=f()}})(function(){var define,module,exports;return function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s}({1:[function(require,module,exports){var BKDRHash=function(str){var seed=131;var seed2=137;var hash=0;str+="x";var MAX_SAFE_INTEGER=parseInt(9007199254740991/seed2);for(var i=0;i<str.length;i++){if(hash>MAX_SAFE_INTEGER){hash=parseInt(hash/seed2)}hash=hash*seed+str.charCodeAt(i)}return hash};module.exports=BKDRHash},{}],2:[function(require,module,exports){var BKDRHash=require("./bkdr-hash");var RGB2HEX=function(RGBArray){var hex="#";RGBArray.forEach(function(value){if(value<16){hex+=0}hex+=value.toString(16)});return hex};var HSL2RGB=function(H,S,L){H/=360;var q=L<.5?L*(1+S):L+S-L*S;var p=2*L-q;return[H+1/3,H,H-1/3].map(function(color){if(color<0){color++}if(color>1){color--}if(color<1/6){color=p+(q-p)*6*color}else if(color<.5){color=q}else if(color<2/3){color=p+(q-p)*6*(2/3-color)}else{color=p}return Math.round(color*255)})};function isArray(o){return Object.prototype.toString.call(o)==="[object Array]"}var ColorHash=function(options){options=options||{};var LS=[options.lightness,options.saturation].map(function(param){param=param||[.35,.5,.65];return isArray(param)?param.concat():[param]});this.L=LS[0];this.S=LS[1];if(typeof options.hue==="number"){options.hue={min:options.hue,max:options.hue}}if(typeof options.hue==="object"&&!isArray(options.hue)){options.hue=[options.hue]}if(typeof options.hue==="undefined"){options.hue=[]}this.hueRanges=options.hue.map(function(range){return{min:typeof range.min==="undefined"?0:range.min,max:typeof range.max==="undefined"?360:range.max}});this.hash=options.hash||BKDRHash};ColorHash.prototype.hsl=function(str){var H,S,L;var hash=this.hash(str);if(this.hueRanges.length){var range=this.hueRanges[hash%this.hueRanges.length];var hueResolution=727;H=hash/this.hueRanges.length%hueResolution*(range.max-range.min)/hueResolution+range.min}else{H=hash%359}hash=parseInt(hash/360);S=this.S[hash%this.S.length];hash=parseInt(hash/this.S.length);L=this.L[hash%this.L.length];return[H,S,L]};ColorHash.prototype.rgb=function(str){var hsl=this.hsl(str);return HSL2RGB.apply(this,hsl)};ColorHash.prototype.hex=function(str){var rgb=this.rgb(str);return RGB2HEX(rgb)};module.exports=ColorHash},{"./bkdr-hash":1}]},{},[2])(2)});
```

Comment:
Would you please let us know how did you generate this minified file?
Many thanks.

Reply:
Hi @sufuf3, I am using uglify-js to generate this js minified file. Could you check the minified file is correct or not? 
And one question what do you mean package.json file meet ^M in each line of the end?  Could you teach me how to remove it?

Reply:
I can tell you how to fix it.
But first of all, would you please let me know what OS you use?
Let's use git commands locally, what do you think?
If you use Unix-like OS, you can try this script https://github.com/sufuf3/cdnjs_setenv or refer to https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md to setup the environment.

Thank you.

Reply:
I am using Windows10, by reading the sparseCheckout.md, for ".git/info/sparse-checkout" file, in my case it should include these lines in below, is that correct?

/ajax/libs/color-hash/*
/build
/documents
/test
/spec
/tools
/LICENSE
/hooks
/*.yml
/*.md
/*.json
/*.sh
/*.js
/.jscsrc
/ajax/libs/color-hash/package.json

Reply:
Would you suggest that I close this pull request and redo it via local git command? @sufuf3

Reply:
@sufuf3 Could you teach me how to send another pull request via git command locally?